node/src/components/consensus/highway_core/state.rs

```diff
@@ -315,14 +322,13 @@ impl<C: Context> State<C> {
         unit: &C::Hash,
         v_ids: I,
     ) -> bool {
-        self.incomplete_endorsements
-            .get(unit)
-            .map(|v| {
-                v_ids
-                    .into_iter()
-                    .all(|v_id| v.get(v_id.0 as usize).is_some())
-            })
-            .unwrap_or(false)
+        if let Some(sigs) = self.endorsements.get(unit) {
+            v_ids.into_iter().all(|v_id| sigs[*v_id].is_some())
```

Comment:
Do we need to iterate? If `unit` is already in `state.endorements` then it is guaranteed to be endorsed. We don't have to add new endorsements to it (validator from `v_ids` we haven't seen yet) and we don't do that in `add_endorsements` either.

Reply:
Good point, yes!